joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/querying/TypeDeclTests.scala

```diff
@@ -350,4 +350,158 @@ class TypeDeclTests extends PhpCode2CpgFixture {
+          val List(anonClass0Param1: Identifier, anonClass0Param2: Literal) = constructAnonClass0.argument.l: @unchecked
+          anonClass0Param1.code shouldBe "$Test0.php:<global>@tmp-0"
+          anonClass0Param2.code shouldBe "10"
+
+          constructAnonClass1.code shouldBe "Test0.php:<global>@anon-class-1->__construct(30)"
+          val List(anonClass1Param1: Identifier, anonClass1Param2: Literal) = constructAnonClass1.argument.l: @unchecked
+          anonClass1Param1.code shouldBe "$Test0.php:<global>@tmp-1"
+          anonClass1Param2.code shouldBe "30"
+        case xs => fail(s"Expected two construct calls, got ${xs.code.mkString("[", ",", "]")}")
+      }
+    }
+
+    "generate `alloc` calls and assignments" in {
+      inside(cpg.method.name("<global>").body.astChildren.isBlock.l) {
+        case constructBlock1 :: constructBlock2 :: Nil =>
+          inside(constructBlock1.astChildren.assignment.l) {
+            case allocAssignment :: Nil =>
+              val Seq(allocTarget: Identifier, allocSource: Call) =
+                List(allocAssignment.target, allocAssignment.source): @unchecked
+              allocTarget.code shouldBe "$Test0.php:<global>@tmp-0"
+              allocSource.code shouldBe "Test0.php:<global>@anon-class-0.<alloc>()"
+              allocSource.methodFullName shouldBe Operators.alloc
+              allocTarget.typeFullName shouldBe "anon-class-0"
+            case xs => fail(s"Expected one assignment, got ${xs.code.mkString("[", ",", "]")}")
+          }
+
+          inside(constructBlock2.astChildren.assignment.l) {
+            case allocAssignment :: Nil =>
+              val Seq(allocTarget: Identifier, allocSource: Call) =
+                List(allocAssignment.target, allocAssignment.source): @unchecked
+              allocTarget.code shouldBe "$Test0.php:<global>@tmp-1"
+              allocSource.code shouldBe "$Test0.php:<global>@anon-class-1.<alloc>()"
+              allocSource.methodFullName shouldBe Operators.alloc
+              allocTarget.typeFullName shouldBe "anon-class-1"
+            case xs => fail(s"Expected some things")
+          }
+
+        case xs => fail(s"Expected one assignment, got ${xs.code.mkString("[", ",", "]")}")
+      }
+    }
+  }
+
+  "Anonymous class nested in class" in {
+    val cpg = code("""<?php
+        |class C {
+        |  function D() {
+        |     new class("foo") {
+        |       private int $x;
+        |     }
+        |  }
+        |}
+        |""".stripMargin)
+
+    inside(cpg.typeDecl.name("C->D@anon-class-\\d+").l) {
+      case anonClass :: Nil =>
+        anonClass.fullName shouldBe s"C->D@anon-class-0"
+        anonClass.member.code.l shouldBe List("$x")
+      case xs => fail(s"Expected one anonymous class, got ${xs.code.mkString("[", ",", "]")}")
+    }
+
+    inside(cpg.method.name("D").body.astChildren.l) {
+      case (localNode: Local) :: (bodyBlock: Block) :: Nil =>
+        localNode.code shouldBe "$C->D@tmp-0"
```

Comment:
Sorry my last comment was not quite correct/clear. 
The temp variables cannot be just numbered via an incremented suffix on a per file basis. So the suffix needs to be "per scope" but since the locals are locals and do not need a globally unique name, it is ok to just name them "tmp0" and so on.

Reply:
From my understanding this `local` node's name is based on the name of the identifier in the method body so I don't think we should modify this in any way as this links to the identifier in the method. From my understanding this local node is being generated in the `LocalCreationPass`

![image](https://github.com/user-attachments/assets/81d3ffa3-7416-4233-a756-906eb8815bac)

Reply:
Yes, the name of a local an the name of the corresponding identifiers should be the same.
Please add tests for the name.
From the graph picture i cannot see what is name and what is code.
For now the local names can stay in this full name style but we will need to change that once the local generation is reworked.